### DISCUSSION TITLE
Battery level should be 'green' through to 100% for LFP battery Model 3s

### DISCUSSION THREAD

**Discussion by CharlesElliottGH** [teslamate-org/teslamate#2595] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

The Battery Level gauge on Overview, as well as Charging Stats and Charge Level screens have 80% as the optimum maximum charge. 

### Expected Behavior

On Tesla Model 3 late 2021> cars, the recommendation from Tesla is to charge to 100%, so the charge should be considered 'Green' through to 100%.

### Steps To Reproduce

View Overview, Charging Stats and Charge Level dashboards which all have green thresholds for charge level at max 80%.

### Relevant log output

```shell
N/A
```

### Type of installation

Docker

### Version

2.9.0

**Comment by LarsAC** [comment]
+1

Maybe we can have per-car battery levels in the settings ?

**Comment by cwanja** [comment]
Do you know if there is any way to identify a LFP battery in the owners API response?

**Comment by LarsAC** [comment]
[VIN Decoder](https://tesladriver.net/tesla-vehicle-vin-number-decoder/) is claiming that digit 7 of the VIN identifies the battery type:

```
Digit 7: Type of drive train
E = Lithium Ion Battery – Electric (NMC)
F = Lithium Iron Phosphate Battery (LFP)
H = Lithium Ion Battery- High Capacity (NMC)
S = Lithium Ion Battery – Standard (NMC)
V = Lithium Ion Battery – Ultra High Capacity (NMC)
```

Would that help ?

**Comment by CharlesElliottGH** [comment]
My 2022 Model 3 has F as the 7th digit, so that seems to work.

**Comment by woyteck1** [comment]
I cannot agree with this one - mine (2021 M3 SR+) also has 7th letter as 'F', but OS clearly shows Daily & Trip charging ranges.

**Comment by CharlesElliottGH** [comment]
A manual setting would be fine for this for now I suggest - if there is a definitive way of finding this via the API in future then it could be changed to trigger from that.

**Comment by woyteck1** [comment]
An update to what I mentioned earlier.
After the most recent update 2022.12.3.2, my charging screen changed to the one that is for LFP batteries. 
I have called Tesla support and apparently my car does have an LFP battery! (It wasn't showing that for last 7 months thou...)
So - what the Digit 7 on VIN may suggest is correct! 

Very odd TBH (the stuff with LFP not showing).

**Reply by manikoutai** [reply]
I think the 7th letter of the VIN is only valid for cars build in China, at least not valid for cars build in north America. I received my LFP car in Jan 2022 and the 7th letter is E.

**Comment by CrowMash** [comment]
In addition to that, could it be possible to set a max value for the "Charging kW" panel depending on car model? Tesla Model 3 (2022) are limited to 170kW for example if I'm correct.

**Comment by OttoKorrect** [comment]
With the recommendation to charge to 100% at least once a week,  I added a days since 100% charge, to my Charge Stats page

![image](https://user-images.githubusercontent.com/7450377/197426670-9e8c6964-c2b9-4bac-9069-2d5d7a060d90.png)

The query is simple:
```
SELECT
	age(max(end_date)::date) as last_full_charge
FROM
	charging_processes
WHERE
	end_battery_level = 100
```

**Reply by OttoKorrect** [reply]
Also my Model Y conforms to the VIN details posted above 
LRWYHC**F**S0NC4*****

**Comment by swiffer** [comment]
VIN decoder is no longer trustworthy as far as i know. see here for ideas on how to fix this.

https://github.com/teslamate-org/teslamate/issues/3524#issuecomment-1850452981
